_blog/2022-12-23-procedural-macros.mdx

```diff
@@ -0,0 +1,417 @@
+---
+title: Reflection in Rust with procedural macros
+description: A tutorial on building and deploying an interactive bot in Rust with Serenity & shuttle
+author: ben
+tags: [rust, tutorial]

```

Comment:
Should we add `macros`, `proc-macros` or `reflection`?

Reply:
Yep, wasn't sure about the tags. I think it is only relevant if there are others in this category and those might be a bit niche. Maybe a `comparison` tag might be good here and might apply to other posts?